test/libyul/yulOptimizerTests/loadResolver/simple_memory.yul

```diff
@@ -4,13 +4,16 @@
     let q := mload(calldataload(0))
     sstore(t, q)
 }
+// ====
+// EVMVersion: >=shanghai
 // ----
 // step: loadResolver
 //
 // {
 //     {
 //         let _2 := calldataload(10)
 //         mstore(calldataload(0), _2)
-//         sstore(mload(_2), _2)
+//         let t := mload(_2)
+//         sstore(t, mload(calldataload(0)))
```

Comment:
Ultimately, this might lead back to an issue in https://github.com/ethereum/solidity/pull/13702 resulting in the data flow analyzer not properly functioning in these cases...